crates/core/src/factories/mem_op_store.rs

```diff
@@ -207,16 +215,21 @@ impl OpStore for Kitsune2MemoryOpStore {
             let self_lock = self.read().await;
 
             let mut used_bytes = 0;
+            let mut candidate_ops = self_lock
+                .op_list
+                .iter()
+                .filter(|(_, op)| {
+                    let loc = op.op_id.loc();
+                    op.created_at >= start
+                        && op.created_at < end
+                        && arc.contains(loc)
+                })
+                .collect::<Vec<_>>();
+            candidate_ops.sort_by_key(|a| a.1.created_at);
```

Comment:
We should return these in order. I actually think this matters to the DHT implementation if the hash map order isn't consistent so this would have been a significant bug if the real op store was implemented this way. For now, it was just about which ops we see in tests which appears a bit random if it's the hashmap order. Good to change this for both cases I believe.